File: language/runtime/apple/core/Sources/RuntimeHostAppleCore/Module/Permission/PermissionRequests.swift
import Foundation

/// The permission request surface attached to one Apple runtime host.
@MainActor
public protocol PermissionRequests {
  /// Submit one permission request to the Apple host.
  func submitPermissionRequest(_ request: RuntimeHostPermissionRequest)

  /// Open the native permission settings surface through the Apple host.
  func openPermissionSettings() -> UInt32
}

File: language/runtime/android/kotlin/src/main/kotlin/dev/destack/runtime/android/module/permission/PermissionRequests.kt
package dev.destack.runtime.android.module.permission

/**
 * The permission request surface attached to one Android runtime host.
 */
public interface PermissionRequests {
    /**
     * Submit one permission request to the Android host.
     */
    public fun submitPermissionRequest(
        request: RuntimeHostPermissionRequest,
    )

    /**
     * Open the native permission settings surface through the Android host.
     */
    public fun openPermissionSettings(): Int
}
